components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/state/OAuthRequestStateValidator.java

```diff
@@ -176,6 +177,10 @@ private void validateInputParameters(OAuthMessage oAuthMessage) throws InvalidRe
                     OAuth2ErrorCodes.INVALID_REQUEST, OAuth2ErrorCodes.OAuth2SubErrorCodes.INVALID_CLIENT);
         }
 
+        String responseType = oAuthMessage.getRequest().getParameter(Constants.RESPONSE_TYPE);
+        if (StringUtils.equals(responseType, Constants.RESPONSE_TYPE_DEVICE)) {
+            return;
```

Comment:
Why are we returning here?
Aren't the other validations applicable for device flow?